sympy/printing/codeprinter.py

```diff
@@ -134,8 +135,19 @@ def _handle_assign_to(expr, assign_to):
                         type(self).__name__, type(assign_to)))
             return Assignment(assign_to, expr)
 
+        #for codeprinting, all lists should be converted to abstract_nodes.Lists.
+        from sympy.codegen.abstract_nodes import List
+        store_list_converter=converter.get(list, None)
+        converter[list]=lambda x: List(*x)
+
         expr = _handle_assign_to(expr, assign_to)
-        expr = _convert_python_lists(expr)
```

Comment:
This is where the explicit conversion was done previously.

Reply:
It needs to happen before `_handle_assign_to` (since it will gladly convert `([],)` to `Tuple([])` in `sympify` which should not be allowed).

Reply:
can we simply move the call up?

Reply:
Currently `_convert_python_lists` only goes through `list`s (so it would stop as soon as it sees the `tuple`). I could try to change it to loop through `tuple` too.

Reply:
I updated `_convert_python_lists` to recurse through `tuple` and moved it forward. I believe this solves all issues.